public_html/wp-content/themes/pattern-directory/functions.php

```diff
@@ -71,14 +71,32 @@ function enqueue_assets() {
 
 		wp_set_script_translations( 'wporg-pattern-script', 'wporg-patterns' );
 
+		wp_add_inline_script(
+			'wporg-pattern-script',
+			sprintf( "var wporgLocale = '%s';", wp_json_encode( get_locale() ) ),
+			'before'
+		);
+
+		wp_add_inline_script(
+			'wporg-pattern-script',
+			sprintf(
+				"var wporgPatternsData = JSON.parse( decodeURIComponent( '%s' ) )",
+				rawurlencode( wp_json_encode( array(
+					'userId' => get_current_user_id(),
+				) ) ),
```

Comment:
Just curious what the purpose of `rawurlencode` is on this one...

Reply:
I thought it was best practice for passing back arrays/objects from PHP to JS, but maybe that's an outdated practice?

Reply:
Oh it could be, though I haven't seen it before. But that's why I asked 😄